packages/fx-core/src/component/generator/apiSpec/helper.ts

```diff
@@ -823,7 +823,12 @@ async function updateAdaptiveCardForCustomApi(
 
     for (const item of specItems) {
       const name = item.item.operationId!.replace(/[^a-zA-Z0-9]/g, "_");
-      const [card] = AdaptiveCardGenerator.generateAdaptiveCard(item.item, true);
+      const [card, jsonPath] = AdaptiveCardGenerator.generateAdaptiveCard(item.item, true);
+      card.body.forEach((value) => {
```

Comment:
You should only replace $data in root level, child $data should not be replaced.